src/scripts/crank.ts

```diff
@@ -198,33 +204,72 @@ async function run() {
           limit: consumeEventsLimit,
           programId: serumProgramId,
         });
-        const transaction = new Transaction({
-          ...recentBlockhash,
-        });
-        transaction.add(
-          ComputeBudgetProgram.setComputeUnitLimit({
-            units: CuLimit,
-          })
-        );
-        if (events.length > priorityQueueLimit) {
-          const isHighFeeMarket = priorityMarkets.includes(i);
-          transaction.add(
-            ComputeBudgetProgram.setComputeUnitPrice({
-              microLamports: isHighFeeMarket ? priorityCuPrice : defaultPriorityCuPrice,
-            })
-          );
+
+        crankInstructionsQueue.push(instr);
+
+        //if the queue is large then add the priority fee
+        if(events.length > priorityQueueLimit){
+          instructionBumpMap.set(instr,1);
         }
-        transaction.add(instr);
 
-        log.info(`market ${i} sending consume events for ${events.length} events`);
+        //bump transaction fee if market address is included in BUMP_MARKETS env
+        if(priorityMarkets.includes(spotMarkets[i].publicKey.toString())){
+          instructionBumpMap.set(instr,1);
+        }
+
+        log.info(`market ${spotMarkets[i].publicKey} creating consume events for ${events.length} events`);
+
+      }
+
+      //send the crank transaction if there are markets that need cranked
+      if(crankInstructionsQueue.length > 0){
+
+        //chunk the instructions to ensure transactions are not too large
+        let chunkedCrankInstructions: any[] = [];
+        let chunkSize = maxTxInstructions;
+        for (let i = 0; i < crankInstructionsQueue.length; i += chunkSize) {
+          chunkedCrankInstructions.push(crankInstructionsQueue.slice(i, i + chunkSize));
+        }
+
+        chunkedCrankInstructions.forEach(function (transactionInstructions){
+
+          let shouldBumpFee = false;
+          let crankTransaction = new Transaction({... recentBlockhash});
+
+          crankTransaction.add(
+              ComputeBudgetProgram.setComputeUnitLimit({
+                units: (CuLimit * maxTxInstructions),
+              })
+          );
+
+          transactionInstructions.forEach(function (crankInstruction) {
+            //check the instruction for flag to bump fee
+            instructionBumpMap.get(crankInstruction) ? shouldBumpFee = true : null;
+            crankTransaction.add(crankInstruction);
+          });
+
+          if(shouldBumpFee || cuPrice){
+            crankTransaction.add(
```

Comment:
isn't it best to have this as the second instruction (the first being compute limit)? I've never tried this, not sure if it works.

Reply:
not 100% required I believe: https://github.com/solpkr/openbook-cranker/commit/0b7b4f2aa8b23e6a450ac8171278ef0f70d774f3